### PR TITLE
(1167c) Update/tidy Pact tests

### PR DESCRIPTION
**Pull request by yndajas**
## To do

- specify submitted on? why/where does this matter?

## Context

Keeping the API and UI in line with Pact tests is becoming a burden, and we appear to be creating a bunch of test data that we don't need

Related UI PRs:
- https://github.com/ministryofjustice/hmpps-accredited-programmes-ui/pull/377
- https://github.com/ministryofjustice/hmpps-accredited-programmes-ui/pull/379

## Changes in this PR

This updates the Pact states match the latest UI tests and removes data that aren't being used anywhere

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes UI for this change to work...
  - [ ] ... and they been released to production already

### Post-merge

- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt [hunk 1]
```diff
@@ -29,56 +29,47 @@ class PactContractTest {
...
-
-  @State("Referral status can be updated")
-  fun `ensure referral status can be updated`() {}
-
-  @State("Referral can be submitted")
-  fun `ensure referral can be submitted`() {}
-
-  @State("Referral summaries exist for an organisation with the ID a026cc07-8e0b-40dd-9b66-1b3dacecc63d")
-  fun `ensure referral summaries exist for an organisation with the ID a026cc07-8e0b-40dd-9b66-1b3dacecc63d`() {}
+  @State("Referral summaries exist for organisation a026cc07-8e0b-40dd-9b66-1b3dacecc63d")
```
Comment (Gweaton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt:71]: As mentioned [on the UI PR](https://github.com/ministryofjustice/hmpps-accredited-programmes-ui/pull/377#discussion_r1409012388) let's make this organisation ID `BWN` to match what's in the test data on an offering.

**Review by Gweaton — COMMENTED** [review]
Looks good, just one thing to update

**Review by Gweaton — APPROVED** [review]

**Comment by yndajas** [comment]
[Fix Pact tests](https://trello.com/c/598Vilhi/1167-fix-pact-tests)

**Comment by yndajas** [comment]
[Fix Pact tests](https://trello.com/c/598Vilhi/1167-fix-pact-tests)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt [hunk 2]
```diff
@@ -29,56 +29,47 @@ class PactContractTest {
...
 
-  @State("Course names exist on the API")
-  fun `ensure course names exist on the API`() {}
+  @State("Course participation 0cff5da9-1e90-4ee2-a5cb-94dc49c4b004 exists")
+  fun `ensure course participation 0cff5da9-1e90-4ee2-a5cb-94dc49c4b004 exists`() {}
 
-  @State("A course exists with ID d3abc217-75ee-46e9-a010-368f30282367")
-  fun `ensure a course exists with ID d3abc217-75ee-46e9-a010-368f30282367`() {}
+  @State("Courses d3abc217-75ee-46e9-a010-368f30282367, 28e47d30-30bf-4dab-a8eb-9fda3f6400e8, and 1811faa6-d568-4fc4-83ce-41118b90242e and no others exist")
+  fun `ensure courses d3abc217-75ee-46e9-a010-368f30282367, 28e47d30-30bf-4dab-a8eb-9fda3f6400e8, and 1811faa6-d568-4fc4-83ce-41118b90242e and no others exist`() {}
```
Comment (anandelangovan1) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt:45]: Question: Is this the same as delete all courses?

Comment (Gweaton) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt:45]: No, we only have the ability to delete individual course participations

**Pull request merged**